File: app/src/main/java/com/example/bilancio/MainActivity.java
package com.example.bilancio;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceManager;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Impostazioni dei pulsanti
        TextView statopatrimoniale = findViewById(R.id.button_statopatrimoniale);
        TextView contoeconomico = findViewById(R.id.button_conto_economico);
        TextView cambiaindici = findViewById(R.id.button_indici);
        TextView file_salvati = findViewById(R.id.button_filesalvati);
        TextView menu = findViewById(R.id.menu_main);

        // Imposta il tema di default se non è già stato fatto
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        if (!prefs.contains("theme")) {
            prefs.edit().putString("theme", "system_default").apply();
        }

        // Listener per i pulsanti che avviano le altre Activity
        menu.setOnClickListener(view -> startActivity(new Intent(MainActivity.this, SettingsActivity.class)));
        file_salvati.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, com.example.bilancio.filesalvati.class)));
        cambiaindici.setOnClickListener(view -> startActivity(new Intent(MainActivity.this, indici.class)));
        statopatrimoniale.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, statopatrimoniale.class)));
        contoeconomico.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, contoeconomico.class)));
    }
}

File: app/src/main/java/com/example/bilancio/MainActivity.kt
package com.example.bilancio

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Impostazioni dei pulsanti
        val statopatrimonialee = findViewById<TextView>(R.id.button_statopatrimoniale)
        val contoeconomicoo = findViewById<TextView>(R.id.button_conto_economico)
        val cambiaindici = findViewById<TextView>(R.id.button_indici)
        val file_salvati = findViewById<TextView>(R.id.button_filesalvati)
        val menu = findViewById<TextView>(R.id.menu_main)

        // Imposta il tema di default se non è già stato fatto
        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        if (!prefs.contains("theme")) {
            prefs.edit().putString("theme", "system_default").apply()
        }

        // Listener per i pulsanti che avviano le altre Activity
        menu.setOnClickListener { view: View? ->
            startActivity(
                Intent(
                    this@MainActivity,
                    SettingsActivity::class.java
                )
            )
        }
        file_salvati.setOnClickListener { v: View? ->
            startActivity(
                Intent(
                    this@MainActivity,
                    filesalvati::class.java
                )
            )
        }
        cambiaindici.setOnClickListener { view: View? ->
            startActivity(
                Intent(
                    this@MainActivity,
                    indici::class.java
                )
            )
        }
        statopatrimonialee.setOnClickListener { v: View? ->
            startActivity(
                Intent(
                    this@MainActivity,
                    statopatrimoniale::class.java
                )
            )
        }
        contoeconomicoo.setOnClickListener { v: View? ->
            startActivity(
                Intent(
                    this@MainActivity,
                    contoeconomico::class.java
                )
            )
        }
    }
}